lib/multicell_location/Kconfig

```diff
@@ -141,6 +141,28 @@ config MULTICELL_LOCATION_HOSTNAME
 	  Server hostname to use when connecting to the selected locations service
 
 if MULTICELL_LOCATION_SERVICE_NRF_CLOUD
+choice MULTICELL_LOCATION_SERVICE_NRF_CLOUD_JWT
+	prompt "Choose how JWT is handled"
+	default MULTICELL_LOCATION_SERVICE_NRF_CLOUD_JWT_GENERATED
+	help
+	  Select how JWT is done
+
+config MULTICELL_LOCATION_SERVICE_NRF_CLOUD_JWT_COMPILE_TIME
```

Comment:
Do we want to encourage this behavior?  What is the use case for a compile time JWT?
JWTs really should be generated per request and/or with an appropriate expiration time.

Reply:
I see your point. Api keys are still there for other services as compile time but yeah, in nature different than JWTs.  Do you have security concerns? 

But in our case (with open DKs - that we are having mostly), at least I haven't been able to get JWT in-device nor runtime generation working and this was the most convenient option to get us further into our beef (=location api) (i.e. use the JWT generated once in some device with long/none expiration date).
Do not know if there would be similar case with actual end user, where customer would like to create one JWT for bunch of devices without of tasks of injecting private key to every device etc.

Since you are here @jayteemo , did you see that comment: https://github.com/tokangas/sdk-nrf/blob/dbf78b7e4159947b041d47c755b0aa1cc7205e77/lib/multicell_location/services/nrf_cloud_integration.c#L75

At least in my case, that was a problem because for some reason I'm having multiple teams in nrf cloud (and cannot delete those), and <audience> tag was needed to be given with uuid of the team that is having access rights.

Reply:
> Do you have security concerns?

A user can do whatever they want in the end, but I would prefer not to encourage such behavior :)

> where customer would like to create one JWT for bunch of devices without of tasks of injecting private key to every device etc.

Again, the customer can do whatever they want, but that is not a proper use case.  The customer should either be using unique device keys or a service key (which is meant for proxy requests, cloud-to-cloud, not on a device):
https://docs.nrfcloud.com/Reference/Interacting/REST#jwt-payloads

Feel free to contact me via email/teams if you are unable to get JWTs working.  You should be using the "device key" method from devices, which requires the `aud` claim to be NULL when creating a JWT in the modem.

Reply:
Yeah, you are right, might be that this compile time JWT should be removed at the end but for now, it is enabling our development on our beef. At some later point, I need to use more time on this to get runtime generation working.

But at latest attempt of JWT stuff, note that I was using the format of "nrf-<imei>" as a "device id" (i.e. not using UUID formatted from modem) when generating the JWT by using at commands. With that I had to include team UUID as <audience> to AT-command to get the JWT generated ok.

Reply:
> With that I had to include team UUID as to AT-command to get the JWT generated ok.

Do you mean for the JWT AT command to work correctly or for nRF Cloud to accept your JWT?

The inclusion of the team UUID is for use with Service Keys, [see "Service Key" bullet point](https://docs.nrfcloud.com/Reference/Interacting/REST/#json-web-tokens-jwt).
Service Keys are meant for cloud-to-cloud access, not device-to-cloud access.  You should be using a Device Key.

Alos, be sure that your device is using newer ECC keys.  If your device has old RSA keys (like those shipped with DK/Thingy, that will not work.

Reply:
We've got key generation working and will get rid of compile time JWT once we've been able to take nrf cloud jwt generation into use.